Src/PerformanceCollector/Filtering/DocumentStream.cs

```diff
@@ -0,0 +1,239 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Filtering
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.Implementation.QuickPulse.Helpers;
+
+    /// <summary>
+    /// Represents a concept of a uniquely identifiable set of full telemetry documents that are being reported by the SDK. 
```

Comment:
- imagine there are 10 UXs out there. Each UX can configure a document stream, which is essentially a set of criteria to which  telemetry items must comply in order to be included into the stream. Each UX can configure more than one document stream if it wishes to. QPS will merge those streams into a single configuration and pass it on to the SDK. SDK doesn't have to know anything about individual UXs, it only has to know about individual streams and their criteria. Think of a document stream as of a set of filters, and it will become clear why there are many of them for a single configuration (of which there's only one on the SDK side).
- the reason why there are per-type lists is because of performance concerns (same reason why there are per-type lists of CalculatedMetrics in CollectionConfiguration). We don't want to cast types, we only do that once and then do if's and compile-time generic methods to execute code for particular ITelemetry type. Since we don't expect many more ITelemetry implementations but do expect huge numbers of items every second, this appears to make sense.
- those are neither configurations nor values. Those are objects created based on configurations that are capable of determining if an ITelemetry passes the test specified by this DocumentStream's criteria. The actual values are only stored in accumulators. Same as for CalculatedMetric instances in CollectionConfiguration.